File: maximum-level-sum-of-a-binary-tree/maximum-level-sum-of-a-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int maxLevelSum(TreeNode* root) {
        if(root == NULL){
            return 0;
        }
        int level = 0;
        queue<TreeNode*> q;
        q.push(root);
        int max_sum = INT_MIN;
        int res = 0;

        while(!q.empty()){
            int n = q.size();
            int sum = 0;
            while(n--){
                TreeNode* node = q.front();
                q.pop();
                
                sum += node->val;
                if(node->left){
                    q.push(node->left);
                }
                if(node->right){
                    q.push(node->right);
                }
            }
            level++;
            if(sum > max_sum){
                max_sum = sum;
                res = level;
            }
        }

        return res;
    }
};

File: maximum-level-sum-of-a-binary-tree/maximum-level-sum-of-a-binary-tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def maxLevelSum(self, root):
        if(root == None):
            return 0

        max_sum = float('-inf')
        result = 0

        queue = deque()
        queue.append(root)
        level = 0

        while queue:
            size = len(queue)
            sum = 0

            for _ in range(size):
                curNode = queue.popleft()
                sum += curNode.val

                if curNode.left:
                    queue.append(curNode.left)
                if curNode.right:
                    queue.append(curNode.right)
            
            level += 1

            if sum > max_sum:
                max_sum = sum
                result = level

        return result;